src/neo/SmartContract/Native/Designate/DesignateContract.cs

```diff
@@ -24,20 +25,18 @@ internal DesignateContract()
             Manifest.Features = ContractFeatures.HasStorage;
         }
 
-        internal override void Initialize(ApplicationEngine engine)
-        {
-            foreach (byte role in Enum.GetValues(typeof(Role)))
-            {
-                engine.Snapshot.Storages.Add(CreateStorageKey(role), new StorageItem(new NodeList()));
-            }
-        }
-
         [ContractMethod(0_01000000, CallFlags.AllowStates)]
-        public ECPoint[] GetDesignatedByRole(StoreView snapshot, Role role)
+        public ECPoint[] GetDesignatedByRole(StoreView snapshot, Role role, uint index)
         {
             if (!Enum.IsDefined(typeof(Role), role))
                 throw new ArgumentOutOfRangeException(nameof(role));
-            return snapshot.Storages[CreateStorageKey((byte)role)].GetInteroperable<NodeList>().ToArray();
+            if (index != uint.MaxValue && snapshot.Height + 2 < index)
```

Comment:
`index != uint.MaxValue`

This condition should be removed.

Reply:
Done